File: cash/cash.c
#include <cs50.h>
#include <stdio.h>
int get_change(void);
int calc(int change);
void printcoinstotal(int coins_total);

int main(void)
{
    int change = get_change();     // gets input from user
    int coins_total = calc(change); // gets value of total number of coins
    printcoinstotal(coins_total);   // prints coins needed
}

int get_change(void)
{
    int change;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0);
    return change;
}

int calc(int change)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int coinstotal = 0;

    do
    {
        if (change >= quarter)
        {
            change = change - quarter;
            coinstotal++;
        }
        else if (change >= dime)
        {
            change = change - dime;
            coinstotal++;
        }
        else if (change >= nickel)
        {
            change = change - nickel;
            coinstotal++;
        }
        else
        {
            change = change - penny;
            coinstotal++;
        }
    }
    while (change > 0);
    return coinstotal;
}

void printcoinstotal(int coins_total)
{
    printf("%i\n", coins_total);
}

File: Problem Set 6/sentimental-cash/cash.py
from cs50 import get_float


def main():
    change = round(get_float("Change: "), 2) * 100
    if change < 0:
        print("Change must be a valid interger")
        return main()
    amount = calcamount(change)
    print(f"{amount}")


def calcamount(n):
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1
    counter = 0
    while n >= 0:
        if n >= quarter:
            counter += 1
            n -= quarter
        elif n >= dime:
            counter += 1
            n -= dime
        elif n >= nickel:
            counter += 1
            n -= nickel
        elif n >= penny:
            counter += 1
            n -= penny
        else:
            return counter


main()
